Tasty/Tasty/Domain/TastyRecipes/Scenes/RecipesListScreen/RecipesListInteractor.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  RecipesListInteractor.swift
+//  Tasty
+//
+//  Created by Алена Панченко on 15.02.2023.
+//
+
+import Foundation
+
+/// Интерактор экрана списка рецептов
+final class RecipesListInteractor: RecipesListBusinessLogic, RecipesListDataStore {
+    
+    // MARK: - Private Constants
+    
+    private enum Constants {
+        static let urlString = "https://api.edamam.com/api/recipes/v2?type=public&q=keto&app_id=1e2c3d39&app_key=%2057a71cae5b22ae60b3dadc7423fad742"
+        static let separatorGridSymbol = "#"
+    }
+    // MARK: - Public Properties
+    
+    var presenter: RecipesListPresentationLogic?
+    var worker: RecipesListWorkerProtocol?
+    var recipeId: String?
+    var recipes: [Recipe] = []
+    var imagesData: [Data] = []
+    
+    // MARK: - Private Properties
+    
+    private let urlString = Constants.urlString
+    
+    // MARK: - Initializers
+    
+    init(presenter: RecipesListPresentationLogic, worker: RecipesListWorker) {
+        self.presenter = presenter
+        self.worker = worker
+    }
+    
+    // MARK: - Public Properties
+    
+    func requestRecipesList() async {
+        guard let url = URL(string: urlString),
+              let worker
+        else { return }
+        do {
+            recipes = try await worker.fetchRecipes(from: url)
+        } catch {
+            print(error)
+        }
+        let response = RecipesListModel.RecipesListChanged.Response(recipes: recipes)
+        presenter?.presentRecipesList(response: response)
+    }
+    
+    func requestId(index: Int) {
+        let response = RecipesListModel.IDRecipeDetailsChanged.Response(idUrlString: recipes[index].uri)
+        let uri = response.idUrlString.split(separator: Constants.separatorGridSymbol)
+        let recipeId = String(uri[1])
```

Comment:
```suggestion
        let url = response.idUrlString.split(separator: Constants.separatorGridSymbol)
        let recipeId = String(url[1])
```
1 - магические числа?